tera-templates/macros.html

```diff
@@ -0,0 +1,77 @@
+{# Constructs a title based on the given crate name and version #}
+{% macro doc_title(name, version) %}
+    {%- if name -%}
+        {{ name }} {{ version | default(value="") }} - Docs.rs
+    {%- else -%}
+        Docs.rs
+    {%- endif -%}
+{% endmacro doc_title %}
+
+{#
+    Makes the appropriate JS imports for highlighting
+    * `languages` An array of strings where each is a valid highlight.js language
+#}
+{% macro highlight_js(languages) %}
+    {# Load the highlight script #}
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.4.0/highlight.min.js" type="text/javascript"
+        charset="utf-8"></script>
+
+    {# Load the script for each provided language #}
+    {%- for language in languages -%}
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.4.0/languages/{{ language }}.min.js"
+        type="text/javascript" charset="utf-8"></script>
+    {%- endfor -%}
+
+    {# Activate highlighting #}
+    <script type="text/javascript" charset="utf-8">
+        hljs.initHighlighting();
+    </script>
```

Comment:
Eventually it would be nice to get rid of inline scripts ... but that can be a separate PR.